components/hab/src/cli.rs

```diff
@@ -1113,28 +1116,26 @@ fn sub_pkg_install(feature_flags: FeatureFlag) -> App<'static, 'static> {
         (@arg AUTH_TOKEN: -z --auth +takes_value "Authentication token for Builder")
         (@arg IGNORE_INSTALL_HOOK: --("ignore-install-hook") "Do not run any install hooks")
     );
-    if feature_flags.contains(FeatureFlag::OFFLINE_INSTALL) {
-        sub = sub.arg(Arg::with_name("OFFLINE").help("Install packages in offline mode")
+    sub = sub.arg(Arg::with_name("OFFLINE").help("Install packages in offline mode")
+                                               .hidden(!feature_flags.contains(FeatureFlag::OFFLINE_INSTALL))
```

Comment:
It is not possible to conditionally add an arg with `structopt` at runtime. The best we can do is conditionally hide the arg.